File: src/main/java/com/freya02/botcommands/api/components/event/EntitySelectEvent.java
package com.freya02.botcommands.api.components.event;

import com.freya02.botcommands.api.BContext;
import com.freya02.botcommands.internal.BContextImpl;
import net.dv8tion.jda.api.events.interaction.component.EntitySelectInteractionEvent;

public class EntitySelectEvent extends EntitySelectInteractionEvent {
	private final BContext context;

	public EntitySelectEvent(BContextImpl context, EntitySelectInteractionEvent event) {
		super(event.getJDA(), event.getResponseNumber(), event.getInteraction());

		this.context = context;
	}

	public BContext getContext() {
		return context;
	}
}

File: src/main/java/com/freya02/botcommands/api/components/event/EntitySelectEvent.kt
package com.freya02.botcommands.api.components.event

import com.freya02.botcommands.api.BContext
import com.freya02.botcommands.api.commands.ratelimit.CancellableRateLimit
import net.dv8tion.jda.api.events.interaction.component.EntitySelectInteractionEvent

class EntitySelectEvent internal constructor(
    val context: BContext,
    event: EntitySelectInteractionEvent,
    cancellableRateLimit: CancellableRateLimit
) : EntitySelectInteractionEvent(event.jda, event.responseNumber, event.interaction),
    CancellableRateLimit by cancellableRateLimit